src/test/java/ch/jalu/collectionbehavior/verification/CollectionMutabilityVerifier.java

```diff
@@ -0,0 +1,199 @@
+        ListWithBackingDataModifier listWithBackingDataModifier =
+            listCreator.createListWithBackingDataModifier("a", "b", "c", "d").orElseThrow();
+
+        return dynamicTest("unmodifiable_changeToOriginalStructureReflectedInList", () -> {
+            List<String> list = listWithBackingDataModifier.list();
+            assertThat(list, contains("a", "b", "c", "d"));
+            listWithBackingDataModifier.runBackingDataModifier();
+            assertThat(list, contains("a", "changed", "c", "d"));
+        });
+    }
+
+    private static DynamicTest createTestForImmutabilityIfApplicable(ListCreator listCreator) {
+        ListWithBackingDataModifier listWithBackingDataModifier =
+            listCreator.createListWithBackingDataModifier("a", "b", "c", "d").orElse(null);
+        if (listWithBackingDataModifier == null) {
+            return null;
+        }
+
+        return dynamicTest("immutable_originalElementDoesNotChangeList", () -> {
+            List<String> list = listWithBackingDataModifier.list();
+            assertThat(list, contains("a", "b", "c", "d"));
+            listWithBackingDataModifier.runBackingDataModifier();
+            assertThat(list, contains("a", "b", "c", "d"));
+        });
+    }
+
+    private static DynamicTest createTestForIteratorModificationIfApplicable(ListCreator listCreator,
+                                                                             List<String> list) {
+        if (listCreator.getSizeLimit() == 0) {
+            return null;
+        }
+
+        return dynamicTest("unmodifiable_iterator", () -> {
+            Iterator<?> iterator = list.iterator();
+            iterator.next();
+            assertThrows(UnsupportedOperationException.class, iterator::remove);
+        });
+    }
+
+    private static void verifyCannotBeModifiedByListIterator(ListModificationBehavior mutability,
+                                                             ListCreator listCreator, List<String> list) {
+        ListIterator<String> listIterator = list.listIterator();
+        assertThrows(UnsupportedOperationException.class, () -> listIterator.add("test"));
+
+        if (listCreator.getSizeLimit() > 0) {
+            assertThat(list.size(), greaterThanOrEqualTo(1));
+
+            listIterator.next();
+            assertThrows(UnsupportedOperationException.class, () -> listIterator.remove());
+            if (mutability.throwsOnModification) {
+                assertThrows(UnsupportedOperationException.class, () -> listIterator.set("test"));
+            } else {
+                assertDoesNotThrow(() -> listIterator.set("test"));
+            }
+        }
+    }
+
+    private static void verifyListExceptionBehavior(List<String> listToVerify,
+                                                    ListModificationBehavior expectedBehavior) {
+        new ListModificationVerifier(listToVerify, expectedBehavior).testMethods();
```

Comment:
in the future, might make sense to include the iterator-based assertions in the ListModificationVerifier, esp. since the iterator stuff could be tested on `reversed()` and `subList()`, too

Reply:
issue #33